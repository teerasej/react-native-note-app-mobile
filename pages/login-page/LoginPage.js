import React, { Component } from 'react'
import { View, Text } from 'react-native'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import LoginForm from './LoginForm';
import { Content } from 'native-base';
import actions from '../../redux/actions';
import { NavigationActions } from '../../redux/NavigationActionClass';



export class LoginPage extends Component {

    static navigationOptions = ({ navigation }) => {
        return {
            headerTitle: <Text>Login</Text>
        };
    };

    constructor(props){
        super(props);

        NavigationActions.setNavigator(this.props.navigation);
    }

    onSignIn = (values) => {
        console.log(values);
        this.props.signIn(values.username, values.password);
    }

    render() {
        return (
            <Content padder>
                <LoginForm onSubmit={this.onSignIn} />
            </Content>
        )
    }
}

const mapStateToProps = (state) => ({

})

const mapDispatchToProps = dispatch => {
    return {
        signIn: (username, password) => dispatch(actions.startSignIn(username,password))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage)
