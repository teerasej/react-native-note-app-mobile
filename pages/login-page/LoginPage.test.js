
import React from 'react';
import { LoginPage } from './LoginPage';
import { Provider } from "react-redux";
import configureStore from "./../../redux/store";
import renderer from 'react-test-renderer';

import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
Enzyme.configure({ adapter: new Adapter() })

const store = configureStore();

describe('Name of the group', () => {

    it('renders without crashing', () => {
        const component = shallow(<LoginPage />);
    });

    test('snapshot', () => {
        const component = renderer.create(
            <Provider store={store}>
                <LoginPage />
            </Provider>
        ).toJSON();
        expect(component).toMatchSnapshot()
    });

    test('has LoginForm', () => {
        const component = shallow(<LoginPage />)
        const instance = component.instance();
        expect(instance.find('LoginForm')).toHaveLength(1);
    });


});
