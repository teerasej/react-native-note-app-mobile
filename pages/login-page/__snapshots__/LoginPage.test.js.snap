// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name of the group renders as expected 1`] = `ShallowWrapper {}`;
